packages/report-server/src/reportBuilder/transform/functions/aggregate/aggregations.ts

```diff
@@ -3,75 +3,124 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Row, FieldValue } from '../../../types';
+import { FieldValue } from '../../../types';
 
 const isUndefined = (value: FieldValue): value is undefined => {
   return value === undefined;
 };
 
-const group = (existingRow: Row, field: string, value: FieldValue) => {
-  existingRow[field] = value;
+const filterUndefinedAndNull = (values: FieldValue[]): NonNullable<FieldValue[]> => {
+  const filteredValues: NonNullable<FieldValue[]> = [];
+  values.forEach(value => {
+    if (value !== undefined && value !== null) {
+      filteredValues.push(value);
+    }
+  });
+  return filteredValues;
+};
+
+const checkIsNum = (values: FieldValue[]): number[] => {
+  const filteredUndefinedAndNullValues = filterUndefinedAndNull(values);
+  const checkedValues: number[] = [];
+  filteredUndefinedAndNullValues.forEach(value => {
+    if (typeof value !== 'number') {
+      throw new Error(`Expected number, got '${typeof value}'.`);
+    }
+    checkedValues.push(value);
+  });
+  return checkedValues;
```

Comment:
Similarly here, I'd prefer we didn't accept an array of values but rather we used something like:
```
const assertIsNumber = (value: FieldValue): assert value is number => {
  if (typeof value !== 'number') {
    throw new Error(`Expected number, got '${typeof value}'.`);
  }
}
```
and then later we can do:
```
values.forEach(assertIsNumber);
```